master/buildbot/steps/artifact.py

```diff
@@ -298,6 +362,13 @@ def rsyncWithRetry(step, origin, destination, port=None):
 
     return retryCommandLinuxOS(rsync_command)
 
+def mkDir(step):
+    if _isWindowsSlave(step):
+        return r'C:\cygwin64\bin\mkdir.exe'
```

Comment:
Even if our production setup we have to call mkdir from cygwin explicitly, since the `mkdir` function from cmd takes precedence over `mkdir.exe` in PATH